### PR TITLE
MergeWithPrevious change

### PR DESCRIPTION
**Pull request by WangPerryWPY**
The method of setAdd, setRemove, setAddAll, setRemoveAll, Increment, AddUnique, AddRelation, RemoveRelation can be merged between the same operation and some other operation. Implementation method reference iOS SDK. I do this by Implementing a tool class in "parse_merge.dart" and calling the "mergeWithPrevious" Method in the set method which implemented in parse_base.dart.

### CODE REVIEW

File: lib/src/objects/parse_base.dart [hunk 1]
```diff
@@ -191,13 +191,18 @@ abstract class ParseBase {
         if (_getObjectData()[key] == value) {
           return;
         }
-        if (forceUpdate) {
-          _getObjectData()[key] = value;
-        }
+        /*if (!forceUpdate) {
```
Comment (phillwiggins) [inline lib/src/objects/parse_base.dart]: Please can you remove this commented out code.

**Comment by phillwiggins** [comment]
Great work! 

Will merge once checks have finished

**Comment by WangPerryWPY** [comment]
Yes, I have tested what I changed, and have removed the commented.
Thanks
 
 
------------------ Original ------------------

**Comment by rodrigosmarques** [comment]
@WangPerryWPY 
I did not understand the change.
These methods already work this way today.
Can you explain the advantage of this change over the previous implementation?

**Comment by WangPerryWPY** [comment]
Before my change, if I setAdd('Test', 1); then setAdd('Test', 2); the first operation will be covered by the last. But after my Change, these operations will become setAdd('Test', [1, 2]); just like all the other parse SDK documents. And setRemove, Increment, setAddUnique is the same like this. 
 
------------------ Original ------------------
From: "Rodrigo de Souza Marques"; 
Date: 2019年8月7日(星期三) 晚上6:24
To: "phillwiggins/flutter_parse_sdk"; 
Cc: "WangPerryWPY"; "Mention"; 
Subject: Re: [phillwiggins/flutter_parse_sdk] MergeWithPrevious change (#240)

 

@WangPerryWPY
 I did not understand the change.
 These methods already work this way today.
 Can you explain the advantage of this change over the previous implementation?

**Comment by pastordee** [comment]
Hi the same thing needs to be done with queryBuilder as well, e.g if you say 
.. orderByDescending('online')
.. orderByDescending('last-seen');

to order online first and then last seen, 
only the online(first one) works, the last seen is not in order but on iOS SDK and ANDROID SDK it works

**Comment by rodrigosmarques** [comment]
@pastordee 
I had not tested it.
I'm doing a PR with the fix

**Comment by rodrigosmarques** [comment]
@phillwiggins and @pastordee 
#241

**Pull request merged**